src/main/scala/org/renci/cam/QueryService.scala

```diff
@@ -0,0 +1,236 @@
+    "rdfs" -> "http://www.w3.org/2000/01/rdf-schema#",
+    "go" -> "http://www.geneontology.org/formats/oboInOwl#",
+    "blml" -> "https://w3id.org/biolink/biolinkml/meta/",
+    "bl" -> "https://w3id.org/biolink/vocab/",
+    "MONDO" -> "http://purl.obolibrary.org/obo/MONDO_",
+    "EMAPA" -> "http://purl.obolibrary.org/obo/EMAPA_",
+    "SO" -> "http://purl.obolibrary.org/obo/SO_",
+    "RO" -> "http://purl.obolibrary.org/obo/RO_",
+    "GO" -> "http://purl.obolibrary.org/obo/GO_",
+    "CHEBI" -> "http://purl.obolibrary.org/obo/CHEBI_",
+    "BFO" -> "http://purl.obolibrary.org/obo/BFO_",
+    "obo" -> "http://purl.obolibrary.org/obo/",
+    "NCBIGENE" -> "http://identifiers.org/ncbigene:",
+    "sesame" -> "http://www.openrdf.org/schema/sesame#",
+    "owl" -> "http://www.w3.org/2002/07/owl#",
+    "prov" -> "http://www.w3.org/ns/prov#",
+    "MESH" -> "http://id.nlm.nih.gov/mesh/"
+  )
+
+  def makeHttpClient: UIO[TaskManaged[Client[Task]]] =
+    ZIO.runtime[Any].map { implicit rts =>
+      BlazeClientBuilder[Task](rts.platform.executor.asEC).withConnectTimeout(Duration(3, MINUTES)).resource.toManaged
+    }
+
+  implicit val sparqlJsonDecoder: EntityDecoder[Task, ResultSet] = EntityDecoder[Task, String].flatMapR { jsonText =>
+    Try(ResultSetFactory.fromJSON(IOUtils.toInputStream(jsonText, StandardCharsets.UTF_8))) match {
+      //IntelliJ shows false compile errors here; check with SBT
+      //Requires import of ZIO-Cats interop typeclasses to compile
+      case Success(resultSet) => DecodeResult.success(resultSet)
+      case Failure(e) => DecodeResult.failure(MalformedMessageBodyFailure("Invalid JSON for SPARQL results", Some(e)))
+    }
+  }
+
+  def getNodeTypes(nodes: List[KGSNode]): Map[String, String] = {
+    val nodeTypes = nodes.collect {
+      case (node) if node.`type`.nonEmpty => (node.id, "bl:" + CaseUtils.toCamelCase(node.`type`, true, '_'))
+    }.toMap
+    val newNodeTypes = nodeTypes ++ nodes.flatMap(node => node.curie.map(node.id -> _)).toMap
+    newNodeTypes
+  }
+
+  def runSPARQLSelectQuery(query: String, appConfig: AppConfig): Task[ResultSet] =
+    for {
+      clientManaged <- makeHttpClient
+      _ = logger.debug("query: {}", query)
+      //building a uri from config shouldn't be this verbose...any better way to do this?
+      uri = Uri(
+        scheme = Some(Uri.Scheme.http),
+        authority =
+          Some(Uri.Authority(host = Uri.RegName(appConfig.`sparql-host`), port = Some(appConfig.`sparql-port`))),
+        path = "/blazegraph/sparql"
+      ).withQueryParam("query", query).withQueryParam("format", "json")
+      _ = logger.debug("uri: {}", uri.toString())
+      request = Request[Task](Method.POST, uri)
+        .withHeaders(Accept(MediaType.application.json), `Content-Type`(MediaType.application.json))
+      response <- clientManaged.use(_.expect[ResultSet](request))
+    } yield response
+
+  def run(limit: Int, queryGraph: KGSQueryGraph, appConfig: AppConfig): Task[ResultSet] = {
```

Comment:
I think we should go with the ZIO environment approach for providing appConfig, as done in the server construction method. So this method signature would be:

```scala
def run(limit: Int, queryGraph: KGSQueryGraph): ZIO[Config[AppConfig], Throwable, ResultSet]

or equivalently

def run(limit: Int, queryGraph: KGSQueryGraph): RIO[Config[AppConfig], ResultSet]
```